### PR TITLE
Events to trigger change in status

### PR DESCRIPTION
**Pull request by arinaldi**

### CODE REVIEW

File: accessGitHub.js [hunk 1]
```diff
@@ -1,13 +1,36 @@
+const EventEmitter = require('events');
 const axios = require('axios');
 const colors = require('colors');
 
+class MyEmitter extends EventEmitter {}
```
Comment (ashleymichal) [inline accessGitHub.js]: unless you're actually going to extend this, maybe just create your emitter off the base class.

File: gistUploader.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ function buildPost(dir, files) {
         content,
       };
 
-      filesProcessed++;
+      filesProcessed += 1;
       if (filesProcessed === array.length) {
```
Comment (ashleymichal) [inline gistUploader.js]: you only want to do this once so you should do it outside of your iterator.

**Review by ashleymichal — CHANGES_REQUESTED** [review]

File: accessGitHub.js (hunk 1 above)
Comment (arinaldi) [reply accessGitHub.js]: Done.

File: gistUploader.js (hunk 2 above)
Comment (arinaldi) [reply gistUploader.js]: Done.

File: gistUploader.js [hunk 3]
```diff
@@ -11,25 +11,17 @@ function buildPost(dir, files) {
...
-      post.files[file] = {
-        content,
-      };
-
-      filesProcessed++;
-      if (filesProcessed === array.length) {
-        accessGitHub.postGist(post);
-      }
-    });
+  files.forEach((file) => {
```
Comment (ashleymichal) [inline gistUploader.js]: last refactor: when you see this pattern:
```
const accumulator = {}; // or let accumulator = [];
things.forEach(thing => { ...accumulator, thingName: thing })
```
you probably have a good case for `Array.prototype.reduce()`

**Review by ashleymichal — CHANGES_REQUESTED** [review]

File: gistUploader.js (hunk 3 above)
Comment (arinaldi) [reply gistUploader.js]: Done.

**Review by ashleymichal — APPROVED** [review]

**Pull request merged**